app/controllers/studentPiece.controller.js

```diff
@@ -159,25 +159,45 @@ exports.deleteAll = (req, res) => {
 
 exports.getStudentRepertoire = async (req, res) => {
   await StudentPiece.findAll({
-    include: {
-      model: db.studentInstrument,
-      required: true,
-      include: [
-        {
-          model: db.userRole,
-          as: "studentRole",
-          required: true,
-          include: {
-            model: db.user,
+    include: [
+      { model: db.semester, required: true },
+      {
+        model: db.piece,
+        required: true,
+        include: { model: db.composer, required: true },
+      },
+      { model: db.semester, required: true },
+      {
+        model: db.piece,
+        required: true,
+        include: { model: db.composer, required: true },
+      },
+      {
+        model: db.studentInstrument,
+        required: true,
+
+        include: [
+          { model: db.instrument, required: true },
+          { model: db.instrument, required: true },
+          {
+            model: db.userRole,
+            as: "studentRole",
             required: true,
-            where: {
-              id: { [Op.eq]: req.params.userId },
+            include: {
+              model: db.user,
+              required: true,
+              where: {
+                id: { [Op.eq]: req.params.userId },
+              },
             },
           },
-        },
-        { model: db.instrument, required: true },
-      ],
-    },
+        ],
+      },
+    ],
+    order: [
+      [db.semester, "name", "asc"],
+      [db.studentInstrument, db.instrument, "name", "asc"],
```

Comment:
Do we want to hard code these, or pass these orders by parameter?

Reply:
don't want to fix it now.